precompiles/xcm/src/lib.rs

```diff
@@ -181,19 +182,27 @@ where
         input.expect_arguments(6)?;
 
         // Read arguments and check it
-        let assets: Vec<MultiLocation> = input
-            .read::<Vec<Address>>()?
+        let assets_raw: Vec<_> = input
+            .read::<BoundedVec<Address, GetMaxAssets<Runtime>>>()?
+            .into();
+
+        let assets = assets_raw
```

Comment:
`.into()` doesn't take any generic arguments so there was a type conversion problem in `.into().iter()` for `BoundedVec` where it couldn't understand the type. I could have managed to do it together but it would make the code look a little ugly and a little unreadable so i decided to break it.

thoughts?

Reply:
It's ok.